<?xml version="1.0" encoding="UTF-8"?>
<!-- Generated by org.testng.reporters.JUnitXMLReporter -->
<testsuite ignored="0" hostname="Administrator" failures="47" tests="166" name="testsuite" time="9.834" errors="0" timestamp="2020-06-30T14:54:18 GMT+07:00">
  <testcase classname="testDAO.ChuyenDeDAOTest" name="deleteWithMaCdNullTest" time="0.045">
    <failure type="org.testng.TestException" message="
Method ChuyenDeDAOTest.deleteWithMaCdNullTest()[pri:0, instance:testDAO.ChuyenDeDAOTest@65ae095c] should have thrown an exception of type class java.lang.Exception">
      <![CDATA[org.testng.TestException: 
Method ChuyenDeDAOTest.deleteWithMaCdNullTest()[pri:0, instance:testDAO.ChuyenDeDAOTest@65ae095c] should have thrown an exception of type class java.lang.Exception
at java.base/java.util.ArrayList.forEach(ArrayList.java:1510)
... Removed 24 stack frames]]>
    </failure>
  </testcase> <!-- deleteWithMaCdNullTest -->
  <testcase classname="testDAO.ChuyenDeDAOTest" name="deleteWithMaKhValidTest" time="0.003"/>
  <testcase classname="testDAO.ChuyenDeDAOTest" name="findByIdWithNotFoundTest" time="0.058"/>
  <testcase classname="testDAO.ChuyenDeDAOTest" name="findByIdWithValidModelTest" time="0.014"/>
  <testcase classname="testDAO.ChuyenDeDAOTest" name="insertWithEmptyModelTest" time="0.005">
    <failure type="org.testng.TestException" message="
Method ChuyenDeDAOTest.insertWithEmptyModelTest()[pri:0, instance:testDAO.ChuyenDeDAOTest@65ae095c] should have thrown an exception of type class java.lang.Exception">
      <![CDATA[org.testng.TestException: 
Method ChuyenDeDAOTest.insertWithEmptyModelTest()[pri:0, instance:testDAO.ChuyenDeDAOTest@65ae095c] should have thrown an exception of type class java.lang.Exception
at java.base/java.util.ArrayList.forEach(ArrayList.java:1510)
... Removed 24 stack frames]]>
    </failure>
  </testcase> <!-- insertWithEmptyModelTest -->
  <testcase classname="testDAO.ChuyenDeDAOTest" name="insertWithNullModelTest" time="0.009"/>
  <testcase classname="testDAO.ChuyenDeDAOTest" name="insertWithValidModelTest" time="0.009"/>
  <testcase classname="testDAO.ChuyenDeDAOTest" name="selectTest" time="0.012"/>
  <testcase classname="testDAO.ChuyenDeDAOTest" name="updateWithEmptyModelTest" time="0.003">
    <failure type="org.testng.TestException" message="
Method ChuyenDeDAOTest.updateWithEmptyModelTest()[pri:0, instance:testDAO.ChuyenDeDAOTest@65ae095c] should have thrown an exception of type class java.lang.Exception">
      <![CDATA[org.testng.TestException: 
Method ChuyenDeDAOTest.updateWithEmptyModelTest()[pri:0, instance:testDAO.ChuyenDeDAOTest@65ae095c] should have thrown an exception of type class java.lang.Exception
at java.base/java.util.ArrayList.forEach(ArrayList.java:1510)
... Removed 24 stack frames]]>
    </failure>
  </testcase> <!-- updateWithEmptyModelTest -->
  <testcase classname="testDAO.ChuyenDeDAOTest" name="updateWithNullModelTest" time="0.002"/>
  <testcase classname="testDAO.ChuyenDeDAOTest" name="updateWithValidModelTest" time="0.003"/>
  <testcase classname="testDAO.ThongKeDAOTest" name="getBangDiemMultipleDataTest" time="0.035">
    <failure type="java.lang.NullPointerException">
      <![CDATA[java.lang.NullPointerException
at com.polypro.dao.ThongKeDAO.getBangDiem(ThongKeDAO.java:67)
at testDAO.ThongKeDAOTest.getBangDiemMultipleDataTest(ThongKeDAOTest.java:89)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1510)
... Removed 23 stack frames]]>
    </failure>
  </testcase> <!-- getBangDiemMultipleDataTest -->
  <testcase classname="testDAO.ThongKeDAOTest" name="getBangDiemTest" time="0.014">
    <failure type="java.lang.NullPointerException">
      <![CDATA[java.lang.NullPointerException
at com.polypro.dao.ThongKeDAO.getBangDiem(ThongKeDAO.java:67)
at testDAO.ThongKeDAOTest.getBangDiemTest(ThongKeDAOTest.java:70)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1510)
... Removed 23 stack frames]]>
    </failure>
  </testcase> <!-- getBangDiemTest -->
  <testcase classname="testDAO.ThongKeDAOTest" name="getDiemTheoChuyenDeTest" time="0.024"/>
  <testcase classname="testDAO.ThongKeDAOTest" name="getDiemTheoChuyenDeWithMultipleDataTest" time="0.024"/>
  <testcase classname="testDAO.ThongKeDAOTest" name="getDoanhThuTest" time="0.022"/>
  <testcase classname="testDAO.ThongKeDAOTest" name="getDoanhThuWithMultipleDataTest" time="0.036"/>
  <testcase classname="testDAO.ThongKeDAOTest" name="getNguoiHocTest" time="0.013"/>
  <testcase classname="testDAO.ThongKeDAOTest" name="getNguoiHocWithMultipleDataTest" time="0.018"/>
  <testcase classname="testModel.ChuyenDeTest" name="getHinhTest" time="0.001"/>
  <testcase classname="testModel.ChuyenDeTest" name="getHocPhiTest" time="0.002"/>
  <testcase classname="testModel.ChuyenDeTest" name="getMaCDTest" time="0.001"/>
  <testcase classname="testModel.ChuyenDeTest" name="getMoTaTest" time="0.003"/>
  <testcase classname="testModel.ChuyenDeTest" name="getTenCDTest" time="0.001"/>
  <testcase classname="testModel.ChuyenDeTest" name="getThoiLuongTest" time="0.001"/>
  <testcase classname="testModel.ChuyenDeTest" name="setHinhNullTest" time="0.001">
    <failure type="org.testng.TestException" message="
Method ChuyenDeTest.setHinhNullTest()[pri:0, instance:testModel.ChuyenDeTest@57aa341b] should have thrown an exception of type class java.lang.Exception">
      <![CDATA[org.testng.TestException: 
Method ChuyenDeTest.setHinhNullTest()[pri:0, instance:testModel.ChuyenDeTest@57aa341b] should have thrown an exception of type class java.lang.Exception
at java.base/java.util.ArrayList.forEach(ArrayList.java:1510)
... Removed 24 stack frames]]>
    </failure>
  </testcase> <!-- setHinhNullTest -->
  <testcase classname="testModel.ChuyenDeTest" name="setHinhTest" time="0.001"/>
  <testcase classname="testModel.ChuyenDeTest" name="setHocPhiNegativeTest" time="0.001">
    <failure type="org.testng.TestException" message="
Method ChuyenDeTest.setHocPhiNegativeTest()[pri:0, instance:testModel.ChuyenDeTest@57aa341b] should have thrown an exception of type class java.lang.Exception">
      <![CDATA[org.testng.TestException: 
Method ChuyenDeTest.setHocPhiNegativeTest()[pri:0, instance:testModel.ChuyenDeTest@57aa341b] should have thrown an exception of type class java.lang.Exception
at java.base/java.util.ArrayList.forEach(ArrayList.java:1510)
... Removed 24 stack frames]]>
    </failure>
  </testcase> <!-- setHocPhiNegativeTest -->
  <testcase classname="testModel.ChuyenDeTest" name="setHocPhiTest" time="0.001"/>
  <testcase classname="testModel.ChuyenDeTest" name="setHocPhiZeroTest" time="0.001">
    <failure type="org.testng.TestException" message="
Method ChuyenDeTest.setHocPhiZeroTest()[pri:0, instance:testModel.ChuyenDeTest@57aa341b] should have thrown an exception of type class java.lang.Exception">
      <![CDATA[org.testng.TestException: 
Method ChuyenDeTest.setHocPhiZeroTest()[pri:0, instance:testModel.ChuyenDeTest@57aa341b] should have thrown an exception of type class java.lang.Exception
at java.base/java.util.ArrayList.forEach(ArrayList.java:1510)
... Removed 24 stack frames]]>
    </failure>
  </testcase> <!-- setHocPhiZeroTest -->
  <testcase classname="testModel.ChuyenDeTest" name="setMaCDNullTest" time="0.004">
    <failure type="org.testng.TestException" message="
Method ChuyenDeTest.setMaCDNullTest()[pri:0, instance:testModel.ChuyenDeTest@57aa341b] should have thrown an exception of type class java.lang.Exception">
      <![CDATA[org.testng.TestException: 
Method ChuyenDeTest.setMaCDNullTest()[pri:0, instance:testModel.ChuyenDeTest@57aa341b] should have thrown an exception of type class java.lang.Exception
at java.base/java.util.ArrayList.forEach(ArrayList.java:1510)
... Removed 24 stack frames]]>
    </failure>
  </testcase> <!-- setMaCDNullTest -->
  <testcase classname="testModel.ChuyenDeTest" name="setMaCDTest" time="0.002"/>
  <testcase classname="testModel.ChuyenDeTest" name="setMoTaEmptyTest" time="0.001"/>
  <testcase classname="testModel.ChuyenDeTest" name="setMoTaTest" time="0.005"/>
  <testcase classname="testModel.ChuyenDeTest" name="setTenCDNullTest" time="0.001">
    <failure type="org.testng.TestException" message="
Method ChuyenDeTest.setTenCDNullTest()[pri:0, instance:testModel.ChuyenDeTest@57aa341b] should have thrown an exception of type class java.lang.Exception">
      <![CDATA[org.testng.TestException: 
Method ChuyenDeTest.setTenCDNullTest()[pri:0, instance:testModel.ChuyenDeTest@57aa341b] should have thrown an exception of type class java.lang.Exception
at java.base/java.util.ArrayList.forEach(ArrayList.java:1510)
... Removed 24 stack frames]]>
    </failure>
  </testcase> <!-- setTenCDNullTest -->
  <testcase classname="testModel.ChuyenDeTest" name="setTenCDTest" time="0.001"/>
  <testcase classname="testModel.ChuyenDeTest" name="setThoiLuongNullTest" time="0.002">
    <failure type="org.testng.TestException" message="
Method ChuyenDeTest.setThoiLuongNullTest()[pri:0, instance:testModel.ChuyenDeTest@57aa341b] should have thrown an exception of type class java.lang.Exception">
      <![CDATA[org.testng.TestException: 
Method ChuyenDeTest.setThoiLuongNullTest()[pri:0, instance:testModel.ChuyenDeTest@57aa341b] should have thrown an exception of type class java.lang.Exception
at java.base/java.util.ArrayList.forEach(ArrayList.java:1510)
... Removed 24 stack frames]]>
    </failure>
  </testcase> <!-- setThoiLuongNullTest -->
  <testcase classname="testModel.ChuyenDeTest" name="setThoiLuongTest" time="0.001"/>
  <testcase classname="testDAO.HocVienDAOTest" name="deleteWithMaHvNegativeTest" time="0.019">
    <failure type="org.testng.TestException" message="
Method HocVienDAOTest.deleteWithMaHvNegativeTest()[pri:0, instance:testDAO.HocVienDAOTest@2e140e59] should have thrown an exception of type class java.lang.Exception">
      <![CDATA[org.testng.TestException: 
Method HocVienDAOTest.deleteWithMaHvNegativeTest()[pri:0, instance:testDAO.HocVienDAOTest@2e140e59] should have thrown an exception of type class java.lang.Exception
at java.base/java.util.ArrayList.forEach(ArrayList.java:1510)
... Removed 24 stack frames]]>
    </failure>
  </testcase> <!-- deleteWithMaHvNegativeTest -->
  <testcase classname="testDAO.HocVienDAOTest" name="deleteWithMaKhValidTest" time="0.001"/>
  <testcase classname="testDAO.HocVienDAOTest" name="findByIdWithNotFoundTest" time="0.023"/>
  <testcase classname="testDAO.HocVienDAOTest" name="findByIdWithValidModelTest" time="0.006"/>
  <testcase classname="testDAO.HocVienDAOTest" name="insertWithEmptyModelTest" time="0.001">
    <failure type="org.testng.TestException" message="
Method HocVienDAOTest.insertWithEmptyModelTest()[pri:0, instance:testDAO.HocVienDAOTest@2e140e59] should have thrown an exception of type class java.lang.Exception">
      <![CDATA[org.testng.TestException: 
Method HocVienDAOTest.insertWithEmptyModelTest()[pri:0, instance:testDAO.HocVienDAOTest@2e140e59] should have thrown an exception of type class java.lang.Exception
at java.base/java.util.ArrayList.forEach(ArrayList.java:1510)
... Removed 24 stack frames]]>
    </failure>
  </testcase> <!-- insertWithEmptyModelTest -->
  <testcase classname="testDAO.HocVienDAOTest" name="insertWithNullModelTest" time="0.001"/>
  <testcase classname="testDAO.HocVienDAOTest" name="insertWithValidModelTest" time="0.001"/>
  <testcase classname="testDAO.HocVienDAOTest" name="selectTest" time="0.003"/>
  <testcase classname="testDAO.HocVienDAOTest" name="updateWithEmptyModelTest" time="0.001">
    <failure type="org.testng.TestException" message="
Method HocVienDAOTest.updateWithEmptyModelTest()[pri:0, instance:testDAO.HocVienDAOTest@2e140e59] should have thrown an exception of type class java.lang.Exception">
      <![CDATA[org.testng.TestException: 
Method HocVienDAOTest.updateWithEmptyModelTest()[pri:0, instance:testDAO.HocVienDAOTest@2e140e59] should have thrown an exception of type class java.lang.Exception
at java.base/java.util.ArrayList.forEach(ArrayList.java:1510)
... Removed 24 stack frames]]>
    </failure>
  </testcase> <!-- updateWithEmptyModelTest -->
  <testcase classname="testDAO.HocVienDAOTest" name="updateWithNullModelTest" time="0.001"/>
  <testcase classname="testDAO.HocVienDAOTest" name="updateWithValidModelTest" time="0.002"/>
  <testcase classname="testModel.HocVienTest" name="getDiemTest" time="0.001"/>
  <testcase classname="testModel.HocVienTest" name="getMaHVTest" time="0.0"/>
  <testcase classname="testModel.HocVienTest" name="getMaKHTest" time="0.001"/>
  <testcase classname="testModel.HocVienTest" name="getMaNHTest" time="0.001"/>
  <testcase classname="testModel.HocVienTest" name="setDiemNegativeTest" time="0.001">
    <failure type="org.testng.TestException" message="
Method HocVienTest.setDiemNegativeTest()[pri:0, instance:testModel.HocVienTest@26e412ef] should have thrown an exception of type class java.lang.Exception">
      <![CDATA[org.testng.TestException: 
Method HocVienTest.setDiemNegativeTest()[pri:0, instance:testModel.HocVienTest@26e412ef] should have thrown an exception of type class java.lang.Exception
at java.base/java.util.ArrayList.forEach(ArrayList.java:1510)
... Removed 24 stack frames]]>
    </failure>
  </testcase> <!-- setDiemNegativeTest -->
  <testcase classname="testModel.HocVienTest" name="setDiemTest" time="0.001"/>
  <testcase classname="testModel.HocVienTest" name="setDiemZeroTest" time="0.001">
    <failure type="org.testng.TestException" message="
Method HocVienTest.setDiemZeroTest()[pri:0, instance:testModel.HocVienTest@26e412ef] should have thrown an exception of type class java.lang.Exception">
      <![CDATA[org.testng.TestException: 
Method HocVienTest.setDiemZeroTest()[pri:0, instance:testModel.HocVienTest@26e412ef] should have thrown an exception of type class java.lang.Exception
at java.base/java.util.ArrayList.forEach(ArrayList.java:1510)
... Removed 24 stack frames]]>
    </failure>
  </testcase> <!-- setDiemZeroTest -->
  <testcase classname="testModel.HocVienTest" name="setMaHVNegativeTest" time="0.001">
    <failure type="org.testng.TestException" message="
Method HocVienTest.setMaHVNegativeTest()[pri:0, instance:testModel.HocVienTest@26e412ef] should have thrown an exception of type class java.lang.Exception">
      <![CDATA[org.testng.TestException: 
Method HocVienTest.setMaHVNegativeTest()[pri:0, instance:testModel.HocVienTest@26e412ef] should have thrown an exception of type class java.lang.Exception
at java.base/java.util.ArrayList.forEach(ArrayList.java:1510)
... Removed 24 stack frames]]>
    </failure>
  </testcase> <!-- setMaHVNegativeTest -->
  <testcase classname="testModel.HocVienTest" name="setMaHVTest" time="0.001"/>
  <testcase classname="testModel.HocVienTest" name="setMaHVZeroTest" time="0.001">
    <failure type="org.testng.TestException" message="
Method HocVienTest.setMaHVZeroTest()[pri:0, instance:testModel.HocVienTest@26e412ef] should have thrown an exception of type class java.lang.Exception">
      <![CDATA[org.testng.TestException: 
Method HocVienTest.setMaHVZeroTest()[pri:0, instance:testModel.HocVienTest@26e412ef] should have thrown an exception of type class java.lang.Exception
at java.base/java.util.ArrayList.forEach(ArrayList.java:1510)
... Removed 24 stack frames]]>
    </failure>
  </testcase> <!-- setMaHVZeroTest -->
  <testcase classname="testModel.HocVienTest" name="setMaKHNullTest" time="0.001">
    <failure type="org.testng.TestException" message="
Method HocVienTest.setMaKHNullTest()[pri:0, instance:testModel.HocVienTest@26e412ef] should have thrown an exception of type class java.lang.Exception">
      <![CDATA[org.testng.TestException: 
Method HocVienTest.setMaKHNullTest()[pri:0, instance:testModel.HocVienTest@26e412ef] should have thrown an exception of type class java.lang.Exception
at java.base/java.util.ArrayList.forEach(ArrayList.java:1510)
... Removed 24 stack frames]]>
    </failure>
  </testcase> <!-- setMaKHNullTest -->
  <testcase classname="testModel.HocVienTest" name="setMaKHTest" time="0.001"/>
  <testcase classname="testModel.HocVienTest" name="setMaNHNullTest" time="0.001">
    <failure type="org.testng.TestException" message="
Method HocVienTest.setMaNHNullTest()[pri:0, instance:testModel.HocVienTest@26e412ef] should have thrown an exception of type class java.lang.Exception">
      <![CDATA[org.testng.TestException: 
Method HocVienTest.setMaNHNullTest()[pri:0, instance:testModel.HocVienTest@26e412ef] should have thrown an exception of type class java.lang.Exception
at java.base/java.util.ArrayList.forEach(ArrayList.java:1510)
... Removed 24 stack frames]]>
    </failure>
  </testcase> <!-- setMaNHNullTest -->
  <testcase classname="testModel.HocVienTest" name="setMaNHTest" time="0.001"/>
  <testcase classname="testDAO.KhoaHocDAOTest" name="deleteWithMaKhNullTest" time="0.02">
    <failure type="org.testng.TestException" message="
Method KhoaHocDAOTest.deleteWithMaKhNullTest()[pri:0, instance:testDAO.KhoaHocDAOTest_$$_jvstd96_0@4c03a37] should have thrown an exception of type class java.lang.Exception">
      <![CDATA[org.testng.TestException: 
Method KhoaHocDAOTest.deleteWithMaKhNullTest()[pri:0, instance:testDAO.KhoaHocDAOTest_$$_jvstd96_0@4c03a37] should have thrown an exception of type class java.lang.Exception
at java.base/java.util.ArrayList.forEach(ArrayList.java:1510)
... Removed 24 stack frames]]>
    </failure>
  </testcase> <!-- deleteWithMaKhNullTest -->
  <testcase classname="testDAO.KhoaHocDAOTest" name="deleteWithMaKhValidTest" time="0.002"/>
  <testcase classname="testDAO.KhoaHocDAOTest" name="findByIdWithNontFoundTest" time="0.029"/>
  <testcase classname="testDAO.KhoaHocDAOTest" name="findByIdWithValidModelTest" time="0.038"/>
  <testcase classname="testDAO.KhoaHocDAOTest" name="insertWithEmptyModelTest" time="0.002">
    <failure type="org.testng.TestException" message="
Method KhoaHocDAOTest.insertWithEmptyModelTest()[pri:0, instance:testDAO.KhoaHocDAOTest_$$_jvstd96_0@4c03a37] should have thrown an exception of type class java.lang.Exception">
      <![CDATA[org.testng.TestException: 
Method KhoaHocDAOTest.insertWithEmptyModelTest()[pri:0, instance:testDAO.KhoaHocDAOTest_$$_jvstd96_0@4c03a37] should have thrown an exception of type class java.lang.Exception
at java.base/java.util.ArrayList.forEach(ArrayList.java:1510)
... Removed 24 stack frames]]>
    </failure>
  </testcase> <!-- insertWithEmptyModelTest -->
  <testcase classname="testDAO.KhoaHocDAOTest" name="insertWithNullModelTest" time="0.001"/>
  <testcase classname="testDAO.KhoaHocDAOTest" name="insertWithValidModelTest" time="0.003"/>
  <testcase classname="testDAO.KhoaHocDAOTest" name="selectTest" time="0.004"/>
  <testcase classname="testDAO.KhoaHocDAOTest" name="updateWithEmptyModelTest" time="0.002">
    <failure type="org.testng.TestException" message="
Method KhoaHocDAOTest.updateWithEmptyModelTest()[pri:0, instance:testDAO.KhoaHocDAOTest_$$_jvstd96_0@4c03a37] should have thrown an exception of type class java.lang.Exception">
      <![CDATA[org.testng.TestException: 
Method KhoaHocDAOTest.updateWithEmptyModelTest()[pri:0, instance:testDAO.KhoaHocDAOTest_$$_jvstd96_0@4c03a37] should have thrown an exception of type class java.lang.Exception
at java.base/java.util.ArrayList.forEach(ArrayList.java:1510)
... Removed 24 stack frames]]>
    </failure>
  </testcase> <!-- updateWithEmptyModelTest -->
  <testcase classname="testDAO.KhoaHocDAOTest" name="updateWithNullModelTest" time="0.001"/>
  <testcase classname="testDAO.KhoaHocDAOTest" name="updateWithValidModelTest" time="0.003"/>
  <testcase classname="testModel.KhoaHocTest" name="getGhiChuTest" time="0.0"/>
  <testcase classname="testModel.KhoaHocTest" name="getHocPhiTest" time="0.001"/>
  <testcase classname="testModel.KhoaHocTest" name="getMaCDTest" time="0.001"/>
  <testcase classname="testModel.KhoaHocTest" name="getMaKHTest" time="0.001"/>
  <testcase classname="testModel.KhoaHocTest" name="getMaNVTest" time="0.003"/>
  <testcase classname="testModel.KhoaHocTest" name="getNgayKGTest" time="0.001"/>
  <testcase classname="testModel.KhoaHocTest" name="getNgayTaoTest" time="0.001"/>
  <testcase classname="testModel.KhoaHocTest" name="getThoiLuongTest" time="0.001"/>
  <testcase classname="testModel.KhoaHocTest" name="setGhiChuEmptyTest" time="0.002"/>
  <testcase classname="testModel.KhoaHocTest" name="setGhiChuTest" time="0.001"/>
  <testcase classname="testModel.KhoaHocTest" name="setHocPhiNegativeTest" time="0.0">
    <failure type="org.testng.TestException" message="
Method KhoaHocTest.setHocPhiNegativeTest()[pri:0, instance:testModel.KhoaHocTest@4ae263bf] should have thrown an exception of type class java.lang.Exception">
      <![CDATA[org.testng.TestException: 
Method KhoaHocTest.setHocPhiNegativeTest()[pri:0, instance:testModel.KhoaHocTest@4ae263bf] should have thrown an exception of type class java.lang.Exception
at java.base/java.util.ArrayList.forEach(ArrayList.java:1510)
... Removed 24 stack frames]]>
    </failure>
  </testcase> <!-- setHocPhiNegativeTest -->
  <testcase classname="testModel.KhoaHocTest" name="setHocPhiTest" time="0.001"/>
  <testcase classname="testModel.KhoaHocTest" name="setHocPhiZeroTest" time="0.001">
    <failure type="org.testng.TestException" message="
Method KhoaHocTest.setHocPhiZeroTest()[pri:0, instance:testModel.KhoaHocTest@4ae263bf] should have thrown an exception of type class java.lang.Exception">
      <![CDATA[org.testng.TestException: 
Method KhoaHocTest.setHocPhiZeroTest()[pri:0, instance:testModel.KhoaHocTest@4ae263bf] should have thrown an exception of type class java.lang.Exception
at java.base/java.util.ArrayList.forEach(ArrayList.java:1510)
... Removed 24 stack frames]]>
    </failure>
  </testcase> <!-- setHocPhiZeroTest -->
  <testcase classname="testModel.KhoaHocTest" name="setMaCDNullTest" time="0.001">
    <failure type="org.testng.TestException" message="
Method KhoaHocTest.setMaCDNullTest()[pri:0, instance:testModel.KhoaHocTest@4ae263bf] should have thrown an exception of type class java.lang.Exception">
      <![CDATA[org.testng.TestException: 
Method KhoaHocTest.setMaCDNullTest()[pri:0, instance:testModel.KhoaHocTest@4ae263bf] should have thrown an exception of type class java.lang.Exception
at java.base/java.util.ArrayList.forEach(ArrayList.java:1510)
... Removed 24 stack frames]]>
    </failure>
  </testcase> <!-- setMaCDNullTest -->
  <testcase classname="testModel.KhoaHocTest" name="setMaCDTest" time="0.001"/>
  <testcase classname="testModel.KhoaHocTest" name="setMaKHNullTest" time="0.002">
    <failure type="org.testng.TestException" message="
Method KhoaHocTest.setMaKHNullTest()[pri:0, instance:testModel.KhoaHocTest@4ae263bf] should have thrown an exception of type class java.lang.Exception">
      <![CDATA[org.testng.TestException: 
Method KhoaHocTest.setMaKHNullTest()[pri:0, instance:testModel.KhoaHocTest@4ae263bf] should have thrown an exception of type class java.lang.Exception
at java.base/java.util.ArrayList.forEach(ArrayList.java:1510)
... Removed 24 stack frames]]>
    </failure>
  </testcase> <!-- setMaKHNullTest -->
  <testcase classname="testModel.KhoaHocTest" name="setMaKHTest" time="0.001"/>
  <testcase classname="testModel.KhoaHocTest" name="setMaNVNullTest" time="0.002">
    <failure type="org.testng.TestException" message="
Method KhoaHocTest.setMaNVNullTest()[pri:0, instance:testModel.KhoaHocTest@4ae263bf] should have thrown an exception of type class java.lang.Exception">
      <![CDATA[org.testng.TestException: 
Method KhoaHocTest.setMaNVNullTest()[pri:0, instance:testModel.KhoaHocTest@4ae263bf] should have thrown an exception of type class java.lang.Exception
at java.base/java.util.ArrayList.forEach(ArrayList.java:1510)
... Removed 24 stack frames]]>
    </failure>
  </testcase> <!-- setMaNVNullTest -->
  <testcase classname="testModel.KhoaHocTest" name="setMaNVTest" time="0.001"/>
  <testcase classname="testModel.KhoaHocTest" name="setNgayKGNullTest" time="0.0">
    <failure type="org.testng.TestException" message="
Method KhoaHocTest.setNgayKGNullTest()[pri:0, instance:testModel.KhoaHocTest@4ae263bf] should have thrown an exception of type class java.lang.Exception">
      <![CDATA[org.testng.TestException: 
Method KhoaHocTest.setNgayKGNullTest()[pri:0, instance:testModel.KhoaHocTest@4ae263bf] should have thrown an exception of type class java.lang.Exception
at java.base/java.util.ArrayList.forEach(ArrayList.java:1510)
... Removed 24 stack frames]]>
    </failure>
  </testcase> <!-- setNgayKGNullTest -->
  <testcase classname="testModel.KhoaHocTest" name="setNgayKGTest" time="0.001"/>
  <testcase classname="testModel.KhoaHocTest" name="setNgayTaoNullTest" time="0.001">
    <failure type="org.testng.TestException" message="
Method KhoaHocTest.setNgayTaoNullTest()[pri:0, instance:testModel.KhoaHocTest@4ae263bf] should have thrown an exception of type class java.lang.Exception">
      <![CDATA[org.testng.TestException: 
Method KhoaHocTest.setNgayTaoNullTest()[pri:0, instance:testModel.KhoaHocTest@4ae263bf] should have thrown an exception of type class java.lang.Exception
at java.base/java.util.ArrayList.forEach(ArrayList.java:1510)
... Removed 24 stack frames]]>
    </failure>
  </testcase> <!-- setNgayTaoNullTest -->
  <testcase classname="testModel.KhoaHocTest" name="setNgayTaoTest" time="0.0"/>
  <testcase classname="testModel.KhoaHocTest" name="setThoiLuongNullTest" time="0.0">
    <failure type="org.testng.TestException" message="
Method KhoaHocTest.setThoiLuongNullTest()[pri:0, instance:testModel.KhoaHocTest@4ae263bf] should have thrown an exception of type class java.lang.Exception">
      <![CDATA[org.testng.TestException: 
Method KhoaHocTest.setThoiLuongNullTest()[pri:0, instance:testModel.KhoaHocTest@4ae263bf] should have thrown an exception of type class java.lang.Exception
at java.base/java.util.ArrayList.forEach(ArrayList.java:1510)
... Removed 24 stack frames]]>
    </failure>
  </testcase> <!-- setThoiLuongNullTest -->
  <testcase classname="testModel.KhoaHocTest" name="setThoiLuongTest" time="0.001"/>
  <testcase classname="testDAO.NguoiHocDAOTest" name="deleteWithMaNhNullTest" time="0.018">
    <failure type="org.testng.TestException" message="
Method NguoiHocDAOTest.deleteWithMaNhNullTest()[pri:0, instance:testDAO.NguoiHocDAOTest@49a71302] should have thrown an exception of type class java.lang.Exception">
      <![CDATA[org.testng.TestException: 
Method NguoiHocDAOTest.deleteWithMaNhNullTest()[pri:0, instance:testDAO.NguoiHocDAOTest@49a71302] should have thrown an exception of type class java.lang.Exception
at java.base/java.util.ArrayList.forEach(ArrayList.java:1510)
... Removed 24 stack frames]]>
    </failure>
  </testcase> <!-- deleteWithMaNhNullTest -->
  <testcase classname="testDAO.NguoiHocDAOTest" name="deleteWithMaNhValidTest" time="0.002"/>
  <testcase classname="testDAO.NguoiHocDAOTest" name="findByIdWithNotFoundTest" time="0.023"/>
  <testcase classname="testDAO.NguoiHocDAOTest" name="findByIdWithValidModelTest" time="0.003"/>
  <testcase classname="testDAO.NguoiHocDAOTest" name="insertWithEmptyModelTest" time="0.001">
    <failure type="org.testng.TestException" message="
Method NguoiHocDAOTest.insertWithEmptyModelTest()[pri:0, instance:testDAO.NguoiHocDAOTest@49a71302] should have thrown an exception of type class java.lang.Exception">
      <![CDATA[org.testng.TestException: 
Method NguoiHocDAOTest.insertWithEmptyModelTest()[pri:0, instance:testDAO.NguoiHocDAOTest@49a71302] should have thrown an exception of type class java.lang.Exception
at java.base/java.util.ArrayList.forEach(ArrayList.java:1510)
... Removed 24 stack frames]]>
    </failure>
  </testcase> <!-- insertWithEmptyModelTest -->
  <testcase classname="testDAO.NguoiHocDAOTest" name="insertWithNullModelTest" time="0.001"/>
  <testcase classname="testDAO.NguoiHocDAOTest" name="insertWithValidModelTest" time="0.003"/>
  <testcase classname="testDAO.NguoiHocDAOTest" name="selectByCourseNotFoundTest" time="0.002"/>
  <testcase classname="testDAO.NguoiHocDAOTest" name="selectByCourseTest" time="0.014"/>
  <testcase classname="testDAO.NguoiHocDAOTest" name="selectByKeywordNotFoundTest" time="0.002"/>
  <testcase classname="testDAO.NguoiHocDAOTest" name="selectByKeywordTest" time="0.003"/>
  <testcase classname="testDAO.NguoiHocDAOTest" name="selectTest" time="0.005"/>
  <testcase classname="testDAO.NguoiHocDAOTest" name="updateWithEmptyModelTest" time="0.001">
    <failure type="org.testng.TestException" message="
Method NguoiHocDAOTest.updateWithEmptyModelTest()[pri:0, instance:testDAO.NguoiHocDAOTest@49a71302] should have thrown an exception of type class java.lang.Exception">
      <![CDATA[org.testng.TestException: 
Method NguoiHocDAOTest.updateWithEmptyModelTest()[pri:0, instance:testDAO.NguoiHocDAOTest@49a71302] should have thrown an exception of type class java.lang.Exception
at java.base/java.util.ArrayList.forEach(ArrayList.java:1510)
... Removed 24 stack frames]]>
    </failure>
  </testcase> <!-- updateWithEmptyModelTest -->
  <testcase classname="testDAO.NguoiHocDAOTest" name="updateWithNullModelTest" time="0.001"/>
  <testcase classname="testDAO.NguoiHocDAOTest" name="updateWithValidModelTest" time="0.001"/>
  <testcase classname="testModel.NguoiHocTest" name="getDienThoaiTest" time="0.001"/>
  <testcase classname="testModel.NguoiHocTest" name="getEmailTest" time="0.001"/>
  <testcase classname="testModel.NguoiHocTest" name="getGhiChuTest" time="0.001"/>
  <testcase classname="testModel.NguoiHocTest" name="getGioiTinhTest" time="0.001"/>
  <testcase classname="testModel.NguoiHocTest" name="getHoTenTest" time="0.001"/>
  <testcase classname="testModel.NguoiHocTest" name="getMaNHTest" time="0.001"/>
  <testcase classname="testModel.NguoiHocTest" name="getMaNVTest" time="0.0"/>
  <testcase classname="testModel.NguoiHocTest" name="getNgayDKTest" time="0.001"/>
  <testcase classname="testModel.NguoiHocTest" name="getNgaySinhTest" time="0.001"/>
  <testcase classname="testModel.NguoiHocTest" name="isGioiTinhTest" time="0.0"/>
  <testcase classname="testModel.NguoiHocTest" name="setDienThoaiNullTest" time="0.001">
    <failure type="org.testng.TestException" message="
Method NguoiHocTest.setDienThoaiNullTest()[pri:0, instance:testModel.NguoiHocTest@7103ab0] should have thrown an exception of type class java.lang.Exception">
      <![CDATA[org.testng.TestException: 
Method NguoiHocTest.setDienThoaiNullTest()[pri:0, instance:testModel.NguoiHocTest@7103ab0] should have thrown an exception of type class java.lang.Exception
at java.base/java.util.ArrayList.forEach(ArrayList.java:1510)
... Removed 24 stack frames]]>
    </failure>
  </testcase> <!-- setDienThoaiNullTest -->
  <testcase classname="testModel.NguoiHocTest" name="setDienThoaiTest" time="0.001"/>
  <testcase classname="testModel.NguoiHocTest" name="setEmailNullTest" time="0.0">
    <failure type="org.testng.TestException" message="
Method NguoiHocTest.setEmailNullTest()[pri:0, instance:testModel.NguoiHocTest@7103ab0] should have thrown an exception of type class java.lang.Exception">
      <![CDATA[org.testng.TestException: 
Method NguoiHocTest.setEmailNullTest()[pri:0, instance:testModel.NguoiHocTest@7103ab0] should have thrown an exception of type class java.lang.Exception
at java.base/java.util.ArrayList.forEach(ArrayList.java:1510)
... Removed 24 stack frames]]>
    </failure>
  </testcase> <!-- setEmailNullTest -->
  <testcase classname="testModel.NguoiHocTest" name="setEmailTest" time="0.001"/>
  <testcase classname="testModel.NguoiHocTest" name="setGhiChuEmptyTest" time="0.001"/>
  <testcase classname="testModel.NguoiHocTest" name="setGhiChuTest" time="0.0"/>
  <testcase classname="testModel.NguoiHocTest" name="setGioiTinhFalseTest" time="0.001"/>
  <testcase classname="testModel.NguoiHocTest" name="setGioiTinhTrueTest" time="0.001"/>
  <testcase classname="testModel.NguoiHocTest" name="setHoTenNullTest" time="0.001">
    <failure type="org.testng.TestException" message="
Method NguoiHocTest.setHoTenNullTest()[pri:0, instance:testModel.NguoiHocTest@7103ab0] should have thrown an exception of type class java.lang.Exception">
      <![CDATA[org.testng.TestException: 
Method NguoiHocTest.setHoTenNullTest()[pri:0, instance:testModel.NguoiHocTest@7103ab0] should have thrown an exception of type class java.lang.Exception
at java.base/java.util.ArrayList.forEach(ArrayList.java:1510)
... Removed 24 stack frames]]>
    </failure>
  </testcase> <!-- setHoTenNullTest -->
  <testcase classname="testModel.NguoiHocTest" name="setHoTenTest" time="0.0"/>
  <testcase classname="testModel.NguoiHocTest" name="setMaNHNullTest" time="0.001">
    <failure type="org.testng.TestException" message="
Method NguoiHocTest.setMaNHNullTest()[pri:0, instance:testModel.NguoiHocTest@7103ab0] should have thrown an exception of type class java.lang.Exception">
      <![CDATA[org.testng.TestException: 
Method NguoiHocTest.setMaNHNullTest()[pri:0, instance:testModel.NguoiHocTest@7103ab0] should have thrown an exception of type class java.lang.Exception
at java.base/java.util.ArrayList.forEach(ArrayList.java:1510)
... Removed 24 stack frames]]>
    </failure>
  </testcase> <!-- setMaNHNullTest -->
  <testcase classname="testModel.NguoiHocTest" name="setMaNHTest" time="0.001"/>
  <testcase classname="testModel.NguoiHocTest" name="setMaNVNullTest" time="0.001">
    <failure type="org.testng.TestException" message="
Method NguoiHocTest.setMaNVNullTest()[pri:0, instance:testModel.NguoiHocTest@7103ab0] should have thrown an exception of type class java.lang.Exception">
      <![CDATA[org.testng.TestException: 
Method NguoiHocTest.setMaNVNullTest()[pri:0, instance:testModel.NguoiHocTest@7103ab0] should have thrown an exception of type class java.lang.Exception
at java.base/java.util.ArrayList.forEach(ArrayList.java:1510)
... Removed 24 stack frames]]>
    </failure>
  </testcase> <!-- setMaNVNullTest -->
  <testcase classname="testModel.NguoiHocTest" name="setMaNVTest" time="0.001"/>
  <testcase classname="testModel.NguoiHocTest" name="setNgayDKNullTest" time="0.001">
    <failure type="org.testng.TestException" message="
Method NguoiHocTest.setNgayDKNullTest()[pri:0, instance:testModel.NguoiHocTest@7103ab0] should have thrown an exception of type class java.lang.Exception">
      <![CDATA[org.testng.TestException: 
Method NguoiHocTest.setNgayDKNullTest()[pri:0, instance:testModel.NguoiHocTest@7103ab0] should have thrown an exception of type class java.lang.Exception
at java.base/java.util.ArrayList.forEach(ArrayList.java:1510)
... Removed 24 stack frames]]>
    </failure>
  </testcase> <!-- setNgayDKNullTest -->
  <testcase classname="testModel.NguoiHocTest" name="setNgayDKTest" time="0.002"/>
  <testcase classname="testModel.NguoiHocTest" name="setNgaySinhNullTest" time="0.0">
    <failure type="org.testng.TestException" message="
Method NguoiHocTest.setNgaySinhNullTest()[pri:0, instance:testModel.NguoiHocTest@7103ab0] should have thrown an exception of type class java.lang.Exception">
      <![CDATA[org.testng.TestException: 
Method NguoiHocTest.setNgaySinhNullTest()[pri:0, instance:testModel.NguoiHocTest@7103ab0] should have thrown an exception of type class java.lang.Exception
at java.base/java.util.ArrayList.forEach(ArrayList.java:1510)
... Removed 24 stack frames]]>
    </failure>
  </testcase> <!-- setNgaySinhNullTest -->
  <testcase classname="testModel.NguoiHocTest" name="setNgaySinhTest" time="0.002"/>
  <testcase classname="testDAO.NhanVienDAOTest" name="deleteWithMaCdNullTest" time="0.018">
    <failure type="org.testng.TestException" message="
Method NhanVienDAOTest.deleteWithMaCdNullTest()[pri:0, instance:testDAO.NhanVienDAOTest@5cd61783] should have thrown an exception of type class java.lang.Exception">
      <![CDATA[org.testng.TestException: 
Method NhanVienDAOTest.deleteWithMaCdNullTest()[pri:0, instance:testDAO.NhanVienDAOTest@5cd61783] should have thrown an exception of type class java.lang.Exception
at java.base/java.util.ArrayList.forEach(ArrayList.java:1510)
... Removed 24 stack frames]]>
    </failure>
  </testcase> <!-- deleteWithMaCdNullTest -->
  <testcase classname="testDAO.NhanVienDAOTest" name="deleteWithMaKhValidTest" time="0.001"/>
  <testcase classname="testDAO.NhanVienDAOTest" name="findByIdWithNotFoundTest" time="0.026"/>
  <testcase classname="testDAO.NhanVienDAOTest" name="findByIdWithValidModelTest" time="0.002"/>
  <testcase classname="testDAO.NhanVienDAOTest" name="insertWithEmptyModelTest" time="0.001">
    <failure type="org.testng.TestException" message="
Method NhanVienDAOTest.insertWithEmptyModelTest()[pri:0, instance:testDAO.NhanVienDAOTest@5cd61783] should have thrown an exception of type class java.lang.Exception">
      <![CDATA[org.testng.TestException: 
Method NhanVienDAOTest.insertWithEmptyModelTest()[pri:0, instance:testDAO.NhanVienDAOTest@5cd61783] should have thrown an exception of type class java.lang.Exception
at java.base/java.util.ArrayList.forEach(ArrayList.java:1510)
... Removed 24 stack frames]]>
    </failure>
  </testcase> <!-- insertWithEmptyModelTest -->
  <testcase classname="testDAO.NhanVienDAOTest" name="insertWithNullModelTest" time="0.001"/>
  <testcase classname="testDAO.NhanVienDAOTest" name="insertWithValidModelTest" time="0.002"/>
  <testcase classname="testDAO.NhanVienDAOTest" name="selectTest" time="0.003"/>
  <testcase classname="testDAO.NhanVienDAOTest" name="updateWithEmptyModelTest" time="0.001">
    <failure type="org.testng.TestException" message="
Method NhanVienDAOTest.updateWithEmptyModelTest()[pri:0, instance:testDAO.NhanVienDAOTest@5cd61783] should have thrown an exception of type class java.lang.Exception">
      <![CDATA[org.testng.TestException: 
Method NhanVienDAOTest.updateWithEmptyModelTest()[pri:0, instance:testDAO.NhanVienDAOTest@5cd61783] should have thrown an exception of type class java.lang.Exception
at java.base/java.util.ArrayList.forEach(ArrayList.java:1510)
... Removed 24 stack frames]]>
    </failure>
  </testcase> <!-- updateWithEmptyModelTest -->
  <testcase classname="testDAO.NhanVienDAOTest" name="updateWithNullModelTest" time="0.0"/>
  <testcase classname="testDAO.NhanVienDAOTest" name="updateWithValidModelTest" time="0.001"/>
  <testcase classname="testModel.NhanVienTest" name="getHoTenTest" time="0.0"/>
  <testcase classname="testModel.NhanVienTest" name="getMaNVTest" time="0.0"/>
  <testcase classname="testModel.NhanVienTest" name="getMatKhauTest" time="0.0"/>
  <testcase classname="testModel.NhanVienTest" name="getVaiTroTest" time="0.001"/>
  <testcase classname="testModel.NhanVienTest" name="isVaiTroTest" time="0.0"/>
  <testcase classname="testModel.NhanVienTest" name="setHoTenNullTest" time="0.0">
    <failure type="org.testng.TestException" message="
Method NhanVienTest.setHoTenNullTest()[pri:0, instance:testModel.NhanVienTest@19ccca5] should have thrown an exception of type class java.lang.Exception">
      <![CDATA[org.testng.TestException: 
Method NhanVienTest.setHoTenNullTest()[pri:0, instance:testModel.NhanVienTest@19ccca5] should have thrown an exception of type class java.lang.Exception
at java.base/java.util.ArrayList.forEach(ArrayList.java:1510)
... Removed 24 stack frames]]>
    </failure>
  </testcase> <!-- setHoTenNullTest -->
  <testcase classname="testModel.NhanVienTest" name="setHoTenTest" time="0.001"/>
  <testcase classname="testModel.NhanVienTest" name="setMaNVNullTest" time="0.0">
    <failure type="org.testng.TestException" message="
Method NhanVienTest.setMaNVNullTest()[pri:0, instance:testModel.NhanVienTest@19ccca5] should have thrown an exception of type class java.lang.Exception">
      <![CDATA[org.testng.TestException: 
Method NhanVienTest.setMaNVNullTest()[pri:0, instance:testModel.NhanVienTest@19ccca5] should have thrown an exception of type class java.lang.Exception
at java.base/java.util.ArrayList.forEach(ArrayList.java:1510)
... Removed 24 stack frames]]>
    </failure>
  </testcase> <!-- setMaNVNullTest -->
  <testcase classname="testModel.NhanVienTest" name="setMaNVTest" time="0.001"/>
  <testcase classname="testModel.NhanVienTest" name="setMatKhauNullTest" time="0.001">
    <failure type="org.testng.TestException" message="
Method NhanVienTest.setMatKhauNullTest()[pri:0, instance:testModel.NhanVienTest@19ccca5] should have thrown an exception of type class java.lang.Exception">
      <![CDATA[org.testng.TestException: 
Method NhanVienTest.setMatKhauNullTest()[pri:0, instance:testModel.NhanVienTest@19ccca5] should have thrown an exception of type class java.lang.Exception
at java.base/java.util.ArrayList.forEach(ArrayList.java:1510)
... Removed 24 stack frames]]>
    </failure>
  </testcase> <!-- setMatKhauNullTest -->
  <testcase classname="testModel.NhanVienTest" name="setMatKhauTest" time="0.0"/>
  <testcase classname="testModel.NhanVienTest" name="setVaiTroTrueTest" time="0.001"/>
  <testcase classname="testModel.NhanVienTest" name="setVaiTrofalseTest" time="0.0"/>
</testsuite> <!-- testsuite -->
